### ISSUE TITLE
iperf3 behavior with SIGINT in client side

### ISSUE THREAD

**Issue by shahar3000** [esnet/iperf#479]
the summary line in the client side is not as expected:
iperf3 -s -p 44406
iperf3 -c 192.168.2.1 -p 44406 -i 1 -t 100 -u -b 0
after using SIGINT I get:
**server:**
[  5] local 192.168.2.1 port 44406 connected to 192.168.2.2 port 43406
[ ID] Interval           Transfer     Bandwidth       Jitter    Lost/Total Datagrams
[  5]   0.00-1.00   sec  8.11 MBytes  68.0 Mbits/sec  1.604 ms  14/1052 (1.3%)
[  5]   1.00-2.00   sec  9.51 MBytes  79.8 Mbits/sec  1.338 ms  0/1217 (0%)
[  5]   2.00-3.00   sec  9.38 MBytes  78.6 Mbits/sec  1.091 ms  0/1200 (0%)
[  5]   3.00-4.00   sec  9.51 MBytes  79.8 Mbits/sec  1.113 ms  0/1217 (0%)
[  5]   4.00-5.00   sec  9.54 MBytes  80.0 Mbits/sec  1.368 ms  0/1221 (0%)
[  5]   4.00-5.00   sec  9.54 MBytes  80.0 Mbits/sec  1.368 ms  0/1221 (0%)
[ ID] Interval           Transfer     Bandwidth       Jitter    Lost/Total Datagrams
[  5]   0.00-5.00   sec  0.00 Bytes  0.00 bits/sec  1.076 ms  14/6999 (0.2%)
iperf3: the client has terminated

the 0 bandwidth and transfer are known bugs in server side , but on the client side I get:
**UDP client:**
[ ID] Interval           Transfer     Bandwidth       Total Datagrams
[  4]   0.00-1.00   sec  8.67 MBytes  72.4 Mbits/sec  1110
[  4]   1.00-2.01   sec  9.53 MBytes  79.9 Mbits/sec  1220
[  4]   2.01-3.00   sec  9.38 MBytes  78.7 Mbits/sec  1200
[  4]   3.00-4.00   sec  9.45 MBytes  79.7 Mbits/sec  1210
[  4]   4.00-5.00   sec  9.53 MBytes  80.0 Mbits/sec  1220
[  4]   5.00-5.85   sec  8.12 MBytes  79.9 Mbits/sec  1040
[ ID] Interval           Transfer     Bandwidth       Jitter    Lost/Total Datagrams
[  4]   0.00-5.85   sec  54.7 MBytes  78.4 Mbits/sec  **0.000 ms  0/7000 (0%)**
[  4] Sent 7000 datagrams
iperf3: interrupt - the client has terminated

**the Jitter and Lost/Total Datagrams are always 0**

when running
iperf3 -s -p 44406
iperf3 -c 192.168.2.1 -p 44406 -i 1 -t 100
I get:
**server:**
[ ID] Interval           Transfer     Bandwidth
[  5]   0.00-1.00   sec  6.29 MBytes  52.8 Mbits/sec
[  5]   1.00-2.00   sec  7.91 MBytes  66.3 Mbits/sec
[  5]   2.00-3.00   sec  7.95 MBytes  66.7 Mbits/sec
[  5]   3.00-4.00   sec  7.88 MBytes  66.1 Mbits/sec
[  5]   4.00-5.00   sec  8.10 MBytes  68.0 Mbits/sec
[  5]   4.00-5.00   sec  8.10 MBytes  68.0 Mbits/sec
[ ID] Interval           Transfer     Bandwidth
[  5]   0.00-5.00   sec  0.00 Bytes  0.00 bits/sec                  sender
[  5]   0.00-5.00   sec  41.8 MBytes  70.1 Mbits/sec                  receiver
iperf3: the client has terminated

the zero in the sender row is a known bug but in client side I get:
**TCP client:**
[ ID] Interval           Transfer     Bandwidth       Retr  Cwnd
[  4]   0.00-1.00   sec  6.76 MBytes  56.7 Mbits/sec   23   55.1 KBytes
[  4]   1.00-2.00   sec  7.98 MBytes  66.9 Mbits/sec    0   74.9 KBytes
[  4]   2.00-3.00   sec  8.02 MBytes  67.3 Mbits/sec    0   89.1 KBytes
[  4]   3.00-4.00   sec  8.02 MBytes  67.3 Mbits/sec    0   96.2 KBytes
[  4]   4.00-5.00   sec  8.13 MBytes  68.2 Mbits/sec    0    100 KBytes
[  4]   5.00-5.41   sec  3.31 MBytes  67.6 Mbits/sec    0    103 KBytes
[ ID] Interval           Transfer     Bandwidth       Retr
[  4]   0.00-5.41   sec  42.2 MBytes  65.5 Mbits/sec   23             sender
[  4]   0.00-5.41   sec  **0.00 Bytes  0.00 bits/sec**                  receiver

**the transfer and bandwidth in the receiver row are always 0**

to sum up:
when interrupting iperf in client side with SIGINT I get the following problems
UDP: in the client sum line Jitter and Lost/Total Datagrams always shows 0
TCP: in the client receiver sum line the transfer and bandwith are always 0

**Comment by bmah888** [comment]
I confirm this behavior.

It looks like the current behavior (in `iperf_got_sigend()`) is designed to print out any accumulated results on the client side without requiring any network interaction with the server.  I think that's actually the most conservative (and best) approach, because if for some reason there's a network partition between the client and server, you absolutely need to be able to stop the client without having it talk to the server.  The disadvantage is that you miss some stats from the server.

Stopping on a signal is presumably not the common case.  So while I agree this behavior isn't optimal, there's a reason for it and I don't see that changing.
